tokio/src/net/tcp/stream.rs

```diff
@@ -219,6 +219,91 @@ impl TcpStream {
         Ok(stream)
     }
 
+    /// Check the TCP stream's read readiness state.
+    ///
+    /// The mask argument allows specifying what readiness to notify on. This
+    /// can be any value, including platform specific readiness, **except**
+    /// `writable`. HUP is always implicitly included on platforms that support
+    /// it.
+    ///
+    /// If the stream is not ready for a read then `Poll::Pending` is returned
+    /// and the current task is notified once a new event is received.
+    ///
+    /// The stream will remain in a read-ready state until calls to
+    /// `poll_read` return `Poll::Pending`.
+    ///
+    /// # Panics
+    ///
+    /// This function panics if:
+    ///
+    /// * `ready` includes writable.
+    /// * called from outside of a task context.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use tokio::io;
+    /// use tokio::net::TcpStream;
+    ///
+    /// use futures::future::poll_fn;
+    ///
+    /// #[tokio::main]
+    /// async fn main() -> io::Result<()> {
+    ///     let stream = TcpStream::connect("127.0.0.:8080").await?;
+    ///
+    ///     poll_fn(|cx| {
+    ///         stream.poll_read_ready(cx, mio::Ready::readable())
+    ///     }).await?;
+    ///
+    ///     Ok(())
+    /// }
+    /// ```
+    pub fn poll_read_ready(
+        &self,
+        cx: &mut Context<'_>,
+        mask: mio::Ready,
```

Comment:
I would remove the mask.

Reply:
@carllerche PR has been rebased on to the current master branch. This method has been reimplemented using `Registration::poll_read_ready`.